### PR TITLE
Python 3.7 build from source

### PR DESCRIPTION
**Pull request by isharov**
* Inspired by official Python docker image
  https://github.com/docker-library/python/blob/master/3.7/stretch/slim/Dockerfile

### CODE REVIEW

**Review by michael-bryson — CHANGES_REQUESTED** [review]
I have tried building this locally and it fails with the following error:
"keyserver receive failed: Cannot assign requested address".
I then created a developer build on our Jenkins system and it also failed but with a slightly different error:
"keyserver receive failed: Connection timed out"
http://sou-jenkins2.swinfra.net/job/CAFapi/view/Developer/job/CAFapi~opensuse-python3-image~mb-build-source~CI/

**Comment by isharov** [comment]
@michael-bryson, fixed, please try the new version.

**Comment by michael-bryson** [comment]
@isharov Both local and Jenkins builds are failing with:
"keyserver receive failed: Connection timed out".

**Comment by dermot-hardy** [comment]
I tried it on my own box and I also get the `mgpg: keyserver receive failed: Connection timed out` error. 
 [Here](https://github.com/CAFapi/opensuse-python3-image/files/4067858/build-log.zip) is my local build log in case it helps.

**Comment by isharov** [comment]
I've added multiple failover keyservers, hope it helps.

**Comment by michael-bryson** [comment]
I now get a different error:
```DOCKER> [91mgpg: keybox '/tmp/tmp.mnakgUni1S/pubring.kbx' created

DOCKER> [91mgpg: keyserver receive failed: Cannot assign requested address

DOCKER> Trying new server...
DOCKER> [91mgpg: keyserver receive failed: Connection timed out

DOCKER> Trying new server...
DOCKER> [91mgpg: keyserver receive failed: Cannot assign requested address

DOCKER> Trying new server...
DOCKER> [91mgpg: keyserver receive failed: Connection timed out

DOCKER> Trying new server...
DOCKER> [91mgpg: no valid OpenPGP data found.
gpg: Total number processed: 0

DOCKER> Trying new server...
DOCKER> [91mgpg: keyserver receive failed: Connection timed out

DOCKER> Trying new server...
DOCKER> [91mgpg: Signature made Wed 18 Dec 2019 07:29:49 PM UTC
gpg:                using RSA key 0D96DF4D4110E5C43FBFB17F2D347EA6AA65421D

DOCKER> [91mgpg: Can't check signature: No public key
```

Attached is my local build log.
[mb-build-log-16-01-2020-1.log](https://github.com/CAFapi/opensuse-python3-image/files/4070277/mb-build-log-16-01-2020-1.log)

**Comment by isharov** [comment]
Strange, it's working on my side.
Seems all this GPG stuff is not an option for your env, so I think about plain SHA512 check instead.

**Comment by dermot-hardy** [comment]
In your environment where it is working do you have direct access to the internet without going though a http/s proxy?

**Comment by isharov** [comment]
Yes, exactly.

**Comment by isharov** [comment]
Just replaced GPG validation by SHA512 checksum matching.

**Comment by michael-bryson** [comment]
With the recent changes the project now builds both locally and on Jenkins.

**Comment by dermot-hardy** [comment]
Great.  Should we make this into a multi-stage build?  It seems odd that the python image would contain so many developer tools.  Would it be a good idea to do the build in one stage and then just copy in the built components to a fresh image?

**Comment by isharov** [comment]
It doesn't contain any development tools (`zypper -n rm --clean-deps $DEVEL_DEPS`).
Also, the build artefacts (by `make install` and `get-pip.py`) aren't obviously determined - the trickiest part of multistage build.
Anyway, probably multistage build would be simpler, I'll try it.

**Comment by isharov** [comment]
Added multistage build.

File: src/main/docker/Dockerfile [hunk 1]
```diff
@@ -32,3 +150,5 @@ LABEL Build.Number="$BUILD_NUMBER" \
       Git.Repo="$GIT_REPO" \
       Git.Branch="$GIT_BRANCH" \
       Git.Commit="$GIT_COMMIT"
+
+CMD ["python3"]
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile]: Tagging the image should come last so move this above the `# Tag the image` comment.

File: src/main/docker/Dockerfile [hunk 2]
```diff
@@ -14,11 +14,129 @@
...
+        "pip==$PYTHON_PIP_VERSION" && \
+    pip --version
+
+
+FROM cafapi/opensuse-base:1
+
+COPY --from=builder /usr/local/bin /usr/local/bin
+COPY --from=builder /usr/local/include /usr/local/include
+COPY --from=builder /usr/local/lib /usr/local/lib
+COPY --from=builder /usr/local/share /usr/local/share
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile]: Can this be done with one `COPY` command? (to reduce the number of layers)

**Review by dermot-hardy — APPROVED** [review]

**Review by michael-bryson — APPROVED** [review]

File: src/main/docker/Dockerfile (hunk 1 above)
Comment (isharov) [reply src/main/docker/Dockerfile]: done

File: src/main/docker/Dockerfile (hunk 2 above)
Comment (isharov) [reply src/main/docker/Dockerfile]: done

**Comment by dermot-hardy** [comment]
@isharov Go ahead and merge this or if you prefer for someone else to do it then re-assign the PR to them.  Thanks, Dermot.

**Pull request merged**